pkg/drivers/vmnet/vmnet.go

```diff
@@ -0,0 +1,250 @@
+	return h.macAddress
+}
+
+// Stop terminates sudo, which will terminate vmnet-helper.
+func (h *Helper) Stop() error {
+	log.Info("Stop vmnet-helper")
+	pidfile := h.pidfilePath()
+	pid, err := process.ReadPidfile(pidfile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// No pidfile.
+		return nil
+	}
+	log.Debugf("Terminate sudo (pid=%v)", pid)
+	if err := process.Terminate(pid, "sudo"); err != nil {
+		if err != os.ErrProcessDone {
+			return err
+		}
+		// No process, stale pidfile.
+		if err := os.Remove(pidfile); err != nil {
+			log.Debugf("failed to remove %q: %s", pidfile, err)
+		}
+	}
+	return nil
+}
+
+// Kill both sudo and vmnet-helper by killing the process group.
+func (h *Helper) Kill() error {
+	log.Info("Kill vmnet-helper")
+	pidfile := h.pidfilePath()
+	pid, err := process.ReadPidfile(pidfile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// No pidfile.
+		return nil
+	}
+	exists, err := process.Exists(pid, "sudo")
+	if err != nil {
+		return err
+	}
+	if !exists {
+		// No process, stale pidfile.
+		if err := os.Remove(pidfile); err != nil {
+			log.Debugf("failed to remove %q: %s", pidfile, err)
+		}
+		return nil
+	}
+	log.Debugf("Kill vmnet-helper process group (pgid=%v)", pid)
+	if err := syscall.Kill(-pid, syscall.SIGKILL); err != nil {
+		if err != syscall.ESRCH {
+			return err
+		}
+		// No process, stale pidfile.
+		if err := os.Remove(pidfile); err != nil {
+			log.Debugf("failed to remove %q: %s", pidfile, err)
+		}
+	}
+	return nil
+}
+
+// GetState returns the sudo child process state.
+func (h *Helper) GetState() (state.State, error) {
+	pidfile := h.pidfilePath()
+	pid, err := process.ReadPidfile(pidfile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return state.Error, err
+		}
+		// No pidfile.
+		return state.Stopped, nil
+	}
+	exists, err := process.Exists(pid, "sudo")
+	if err != nil {
+		return state.Error, err
+	}
+	if !exists {
+		// No process, stale pidfile.
+		if err := os.Remove(pidfile); err != nil {
+			log.Debugf("failed to remove %q: %s", pidfile, err)
+		}
+		return state.Stopped, nil
+	}
+	return state.Running, nil
+}
+
+func (h *Helper) openLogfile() (*os.File, error) {
+	logfile := filepath.Join(h.MachineDir, logfileName)
+	return os.OpenFile(logfile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+}
+
+func (h *Helper) pidfilePath() string {
+	return filepath.Join(h.MachineDir, pidfileName)
+}
+
+// Apple recommend receive buffer size to be 4 times the size of the send buffer
+// size, but send buffer size is not used to allocate a buffer in datagram
+// sockets, it only limits the maximum packet size. Must be larger than TSO
+// packets size (65550 bytes).
+const sendBufferSize = 65 * 1024
+
+// The receive buffer size determine how many packets can be queued by the
+// peer. Using bigger receive buffer size make ENOBUFS error less likely for the
+// peer and improves throughput.
```

Comment:
Update the comment from https://github.com/nirs/vmnet-helper/pull/63/commits/2f103e0181d460068560b0ff159e5fdb353a1d51